dhctl/pkg/operations/bootstrap/steps.go

```diff
@@ -159,6 +171,117 @@ func CheckBashibleBundle(sshClient *ssh.Client) bool {
 	return bashibleUpToDate
 }
 
+func SetupSSHTunnelToRegistryPackagesProxy(sshCl *ssh.Client) (*frontend.ReverseTunnel, error) {
+	tun := sshCl.ReverseTunnel("5444:127.0.0.1:5444")
+	err := tun.Up()
+	if err != nil {
+		return nil, err
+	}
+
+	return tun, nil
+}
+
+type registryClientConfigGetter struct {
+	registry.ClientConfig
+}
+
+func newRegistryClientConfigGetter(config config.RegistryData) (*registryClientConfigGetter, error) {
+	auth, err := config.Auth()
+	if err != nil {
+		return nil, fmt.Errorf("registry auth: %v", err)
+	}
+
+	return &registryClientConfigGetter{
+		ClientConfig: registry.ClientConfig{
+			Repository: strings.Join([]string{config.Address, config.Path}, "/"),
+			Scheme:     config.Scheme,
+			CA:         config.CA,
+			Auth:       auth,
+		},
+	}, nil
+}
+
+func (r *registryClientConfigGetter) Get(_ string) (*registry.ClientConfig, error) {
+	return &r.ClientConfig, nil
+}
+
+func StartRegistryPackagesProxy(config config.RegistryData) error {
+	cert, err := generateTLSCertificate()
+	if err != nil {
+		return fmt.Errorf("failed to generate TLS certificate: %v", err)
+	}
+
+	listener, err := tls.Listen("tcp", "127.0.0.1:5444", &tls.Config{
+		Certificates: []tls.Certificate{cert},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to listen: %v", err)
+	}
+
+	logger := logrus.New()
```

Comment:
Please add interface in packages proxy packet, implement in dhctl with log package and pass to proxy

Reply:
done